pkg/hive/manager.go

```diff
@@ -57,44 +60,63 @@ func NewClusterManagerFromConfig(restConfig *rest.Config) (ClusterManager, error
 		return nil, err
 	}
 
-	return newClusterManager(hiveclientset, kubernetescli), nil
+	dh, err := dynamichelper.New(log, restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return newClusterManager(log, hiveclientset, kubernetescli, dh), nil
 }
 
-func newClusterManager(hiveClientset *hiveclient.Clientset, kubernetescli *kubernetes.Clientset) ClusterManager {
+func newClusterManager(log *logrus.Entry, hiveClientset *hiveclient.Clientset, kubernetescli *kubernetes.Clientset, dh dynamichelper.Interface) ClusterManager {
 	return &clusterManager{
 		hiveClientset: hiveClientset,
 		kubernetescli: kubernetescli,
+
+		dh: dh,
 	}
 }
 
-func (hr *clusterManager) Register(ctx context.Context, workloadCluster *WorkloadCluster) (*hivev1.ClusterDeployment, error) {
-	var namespace string
-
+func (hr *clusterManager) CreateNamespace(ctx context.Context) (*corev1.Namespace, error) {
+	var namespaceName string
+	var namespace *corev1.Namespace
 	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		namespace = "aro-" + uuid.Must(uuid.NewV4()).String()
-		csn := ClusterNamespace(namespace)
-		_, err := hr.kubernetescli.CoreV1().Namespaces().Create(ctx, csn, metav1.CreateOptions{})
+		namespaceName = "aro-" + uuid.Must(uuid.NewV4()).String()
```

Comment:
I still don't like this :) I raised a PR so that we can perhaps get the ID we need easier. https://github.com/Azure/ARO-RP/pull/2236

Reply:
I left comments in your PR. Let's continue this discussion there. I'm happy to change this when/if we find a solution for this.

We have time to make this change since, despite being in master already, this code is a no-op for now (no hive in prod at the moment).